.github/workflows/install_deps/action.yml

```diff
@@ -0,0 +1,113 @@
+name: Install Dependencies
+description: |
+  This workflow installs Python, PennyLane and all its dependencies. If a
+  requirements file is provided, it will upload the results of pip freeze
+  as well.
+inputs:
+  python_version:
+    description: The version of Python to use in order to run unit tests
+    required: false
+    default: '3.9'
+  install_jax:
+    description: Indicate if JAX should be installed or not
+    required: false
+    default: 'true'
+  jax_version:
+    description: The version of JAX to install for any job that requires JAX
+    required: false
+    default: 0.4.23
+  install_tensorflow:
+    description: Indicate if TensorFlow should be installed or not
+    required: false
+    default: 'true'
+  tensorflow_version:
+    description: The version of TensorFlow to install for any job that requires TensorFlow
+    required: false
+    default: 2.15.0
+  install_pytorch:
+    description: Indicate if PyTorch should be installed or not
+    required: false
+    default: 'true'
```

Comment:
I couldn't find a way for composite actions specifically to accept anything instead of strings. I tried with bools first and it failed, so I just did this because it worked : )

Reply:
I'm not sure I follow: have you tried adding a [`type: boolean`](https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_callinputsinput_idtype) flag to the input?

I've had success doing this for workflows in an internal project.

Reply:
I had it that way in a past commit:
https://github.com/PennyLaneAI/pennylane/blob/1ed55d3a90e29dd85bcbae0b0a64688d41474a4c/.github/workflows/install_deps/action.yml#L22-L26

...but then the [catalyst tests failed](https://github.com/PennyLaneAI/pennylane/actions/runs/7801565173/job/21276985885) because TF wasn't installed (the tf tests only passed because it skipped them all lol). If you check the "Install PennyLane and Dependencies" section of that run, you can see that `install_tensorflow` is `true`, but it was interpreted as a string, passed in like this:
https://github.com/PennyLaneAI/pennylane/blob/1ed55d3a90e29dd85bcbae0b0a64688d41474a4c/.github/workflows/interface-unit-tests.yml#L350

so it ended up not installing TensorFlow.